### DISCUSSION TITLE
ScriptReference/Highlighter

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#241] [Notes]
# ScriptReference/Highlighter

https://docs.unity3d.com/ScriptReference/Highlighter.html

**Comment by mvi** [comment]
Note that this is not the same as the highlighting in the project settings/preferences window when you type something in search. Instead this puts a box around a rect that is found by matching a name or identifier.  
An easy example is the following:  
  

```csharp
Highlighter.Highlight("Console", "Error Pause");
```
  
![Highlight](https://uploads.disquscdn.com/images/0b5fd74d065011a812f9a055c164f20b7a242cc3f9087320d78ad6c468aeaba6.gif)  
If you are looking for the project settings style you can call `EditorGUI.BeginLabelHighlight` and `EndLabelHighlight` via reflection - source [here](https://github.com/Unity-Technologies/UnityCsReference/blob/master/Editor/Mono/EditorGUI.cs#L6496).

**Comment by docscomments** [comment]
**Comment by `@ta-david-yu`:**  

Also note, only `EditorGUI(Layout).PrefixLabel` will be highlighted when using the internal `EditorGUI.BeginLabelHighlight`.
